controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -193,6 +193,14 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		nb.Spec.DisableLoadBalancerService = sc.Spec.MultiCloudGateway.DisableLoadBalancerService
 	}
 
+	if sc.Spec.MultiCloudGateway != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName != "" {
+		nb.Spec.ExternalPgSSLRequired = true
+		nb.Spec.ExternalPgSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName, Namespace: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretNamespace}
+		nb.Spec.ExternalPgSSLUnauthorized = sc.Spec.MultiCloudGateway.ExternalPgConfig.AllowSelfSignedCerts
+		if sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName != "" && sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretNamespace != "" {
```

Comment:
what if someone sets the TLSSecret but forgets to set PGSecret details, should we error out in that case?

Reply:
I have changed the check above to see if PGSecret is empty or not. It won't enter this block if it is not set.